srcCpp/RTIDDSImpl.cxx

```diff
@@ -2495,6 +2779,369 @@ IMessagingWriter *RTIDDSImpl<T>::CreateWriter(const char *topic_name)
     }
   #endif
 
+  #ifdef RTI_FLATDATA_AVAILABLE
+    if (_isFlatData) {
+        char buf[10];
+        sprintf(buf, "%d", DDS_LENGTH_UNLIMITED);
+        DDSPropertyQosPolicyHelper::add_property(dw_qos.property,
+                "dds.data_writer.history.memory_manager.fast_pool.pool_buffer_max_size",
+                buf,
+                false);
+
+        /**
+         * If FlatData and LargeData, automatically estimate initial_samples here
+         * in a range from 1 up to the initial samples specifies in the QoS file.
+         *
+         * This is done to avoid using too much memory since DDS allocates
+         * samples of the RTI_FLATDATA_MAX_SIZE size
+         */
+        if (_isLargeData) {
+            max_allocable_space = MAX_PERFTEST_SAMPLE_SIZE;
+
+          #ifdef RTI_DARWIN
+            /**
+             * In OSX, we might not be able to allocate all the send queue samples
+             * We only need this on the DW since it will allocate the samples
+             * on Zero Copy
+             */
+            if (_isZeroCopy) {
+                max_allocable_space = getShmemSHMMAX();
+                fprintf(stdout, "SHMMAX: %lu\n", max_allocable_space);
+                //max_allocable_space = MAX_DARWIN_SHMEM_SIZE;
+
+                /**
+                 * Leave enought room for an sceneario of two participants:
+                 *   - One Publisher with one DW (throughput topic)
+                 *   - One Subscriber with two DW (Latency topic and Announcement)
+                 */
+                max_allocable_space /= 3;
```

Comment:
Because it is the only SHMEM resources constrained OS we are supporting. This is done in order to allow perftest running on OSX with low SHMEM resources even though we lose performance. (We still display a message about how to increase these resource limits)

Reply:
:+1: